doc/table.dox

```diff
@@ -158,17 +158,58 @@ namespace lsst { namespace afw { namespace table {
  *  discussed below).
  *
  *  @section afwTableFieldNames Field Names
- *  By convention, field names are all lowercase and have '.'-separated elements.  Only letters, numbers
- *  and periods should be used.  These rules are not enforced, but names that do not meet these requirements
- *  may not round-trip correctly in FITS (periods are converted to underscores in the FITS persistence
- *  layer, so we cannot distinguish between the two when we read tables from FITS).
- *  Schema provides extra functionality for names with period-separated elements; these elements can
- *  be accessed separately individually with the bracket operators.  More information on schema namespaces
- *  can be found in the Schema and SubSchema class documentation, and the testSchema.py unit test may
- *  also be a useful example.
- *
- *  Other field strings (documentation and units) are essentially arbitrary, but should not contain
- *  single quotes, as these may also confuse FITS parsers (even when escaped).
+ *
+ *  By convention, field names are CamelCase and have '_'-separated elements.
+ *  Only letters, numbers and underscores should be used.
+ *  These rules are not enforced, but names that do not meet these requirements
+ *  may not round-trip correctly in FITS.
+ *
+ *  Underscores should be used as a sort of namespace separator (much like '.'
+ *  in Python; we use underscore instead so we don't need to translate field
+ *  names when using them in SQL).  These namespace elements typically indicate
+ *  things like the module or Task containing the algorithm that produced the
+ *  field, the name of the algorithm, and (finally) the name of the particular
+ *  output produced by that algorithm.
+ *
+ *  For example:
+ *
+ *    - `base_SdssShape_xx` is produced by the SdssShape algorithm located in
+ *      the meas_base package (we drop "meas_" since so many algorithms are in
+ *      meas_* packages; see `lsst.meas.base.generateAlgorithmName`), and this
+ *      particular field represents the x-x second-moment of the source.
+ *
+ *    - `deblend_nChild` is produced by `SourceDeblendTask`, where `deblend`
+ *      is an abbreviation of the Task, and `nChild` is a value produced by
+ *      it.  There are no hard rules for how to abbreviate the name of a Task
+ *      when generating its field names; we trust developer judgement in
+ *      selecting a prefix that is both concise and unambiguous.
+ *
+ *  Note that some underscore-separated elements are themselves multiple words,
+ *  such as `SdssShape` or `nChild`, and we use CamelCase, not more underscores,
```

Comment:
(c)CamelCase?

Reply:
I think I'll leave this one and the next as is; I don't want the punctuation to be confusing, either.